Source/ReadyPlayerMe/Private/Storage/AvatarManifest.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright © 2021++ Ready Player Me
+
+
+#include "AvatarManifest.h"
+
+#include "ReadyPlayerMeSettings.h"
+#include "Storage/AvatarStorage.h"
+#include "Utils/AvatarManifestExtractor.h"
+
+void FAvatarManifest::AddAvatar(const FString& AvatarId)
+{
+	if (!AvatarRecords.Contains(AvatarId))
+	{
+		AvatarRecords.Add(AvatarId, FDateTime::Now().ToUnixTimestamp());
+	}
+	EnforceAvatarLimit();
```

Comment:
I'm not sure it's the best approach to force the use of `EnforceAvatarLimit()` here. This functions purpose is to add an avatar (judging by the name), I understand your reasoning here but at the moment this function has a "hidden side effect" of also removing avatars which may be confusing if you actually want to just add an avatar. My suggestion would be that you could just make EnforceAvatarLimit() public so developers can call whenever they like, which enables custom cache management. Additionally, if you still want to have a simplified function that does both, you could also have a separate AddAvatarAndEnforceLimit method that does both of these. Just a suggestion though.

Reply:
+1 to this - if I as a developer called a method to add an avatar and suddenly it started removing stuff I would also be very confused 😅

Reply:
Done the way we agreed on.